src/fidesctl/cli/commands/core.py

```diff
@@ -141,11 +142,18 @@ def parse(ctx: click.Context, manifests_dir: str, verbose: bool = False) -> None
 def sync(ctx: click.Context, manifests_dir: str) -> None:
     """
     Update local resource files by their fides_key to match their server versions.
+
+    Aborts the sync if there are unstaged or untracked files in the manifests dir.
     """
 
     config = ctx.obj["CONFIG"]
     # Do this to validate the manifests since they won't get parsed during the sync process
     _parse.parse(manifests_dir)
+    if git_is_dirty(manifests_dir):
+        echo_red(
+            f"There are unstaged changes in your manifest directory: '{manifests_dir}' \nAborting sync!"
```

Comment:
One possible improvement is to print the output of `git_session.status` here

Reply:
i thought about that but opted instead to only let the user know where the directory is that the changes occur, since the status output can be messy/complicated